### PR TITLE
Add DeploymentConfig to GetDeployMetaFromPod

### PR DESCRIPTION
**Pull request by monsteredc**
**Please provide a description of this PR:**
Related to https://github.com/istio/istio/issues/36155.

@bianpengyuan @lucasponce

### CODE REVIEW

File: pkg/kube/util.go [hunk 1]
```diff
@@ -231,6 +231,9 @@ func GetDeployMetaFromPod(pod *kubeApiCore.Pod) (metav1.ObjectMeta, metav1.TypeM
 				name := strings.TrimSuffix(controllerRef.Name, "-"+pod.Labels["pod-template-hash"])
 				deployMeta.Name = name
 				typeMetadata.Kind = "Deployment"
+			} else if typeMetadata.Kind == "ReplicationController" && pod.Labels["deploymentconfig"] != "" {
+				deployMeta.Name = pod.Labels["deploymentconfig"]
```
Comment (howardjohn) [inline pkg/kube/util.go:235]: what is this 'deploymentconfig' label? I don't see any reference is kubernetes.

**Review by howardjohn — COMMENTED** [review]
Can you add a test case for this?

File: pkg/kube/util.go (hunk 1 above)
Comment (bianpengyuan) [reply pkg/kube/util.go:235]: It is an openshift resource: https://docs.openshift.com/container-platform/4.1/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are We need to comment it well here and point to the openshift reference for this label.

**Comment by bianpengyuan** [comment]
@lucasponce please take a look at this change. thanks!

File: pkg/kube/util.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/kube/util.go:235]: `ReplicationController` was replaced by deployment several years ago. Should be no use anymore

Comment (monsteredc) [reply pkg/kube/util.go:235]: For the openshift platform, pods which are controlled by `ReplicationController` can still be created by `DeploymentConfig`.

**Comment by lucasponce** [comment]
> @lucasponce please take a look at this change. thanks!

Patch looks ok, thanks !

DeploymentConfig are old controllers, but there are a large population of OpenShift users that still use them, and they want to add them to the mesh without migrating, so this patch will help them to get the proper name of the workload.

File: pkg/kube/util.go [hunk 2]
```diff
@@ -231,6 +231,19 @@ func GetDeployMetaFromPod(pod *kubeApiCore.Pod) (metav1.ObjectMeta, metav1.TypeM
...
+				//
+				// nolint: lll
+				// For DeploymentConfig details, refer to
+				// https://docs.openshift.com/container-platform/4.1/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are
+				//
+				// For the reference to the pod label 'deploymentconfig', refer to
+				// https://github.com/openshift/library-go/blob/7a65fdb398e28782ee1650959a5e0419121e97ae/pkg/apps/appsutil/const.go#L25
+				deployMeta.Name = pod.Labels["deploymentconfig"]
+				typeMetadata.Kind = "DeploymentConfig"
+				delete(deployMeta.Labels, "deploymentconfig")
```
Comment (bianpengyuan) [inline pkg/kube/util.go:246]: Why do we delete this label here?

Comment (bianpengyuan) [reply pkg/kube/util.go:246]: Actually it makes sense, since this label comes should only exist at the pod.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**